src/exchange_translator.h

```diff
@@ -78,24 +81,32 @@ class ExchangeTranslator {
   void AddArc(Request<T>* req, Bid<T>* bid, ExchangeGraph::Ptr graph) {
     double pref =
         ex_ctx_->trader_prefs.at(req->requester())[req][bid];
+    // TODO: make the following check `pref <=0` and remove the `else if` block
+    // before release 1.4
     if (pref < 0) {
       CLOG(LEV_DEBUG1) << "Removing arc because of negative preference.";
-    } else {
-      // get translated arc
-      Arc a = TranslateArc(xlation_ctx_, bid);
-
-      a.unode()->prefs[a] = pref;  // request node is a.unode()
-      int n_prefs = a.unode()->prefs.size();
-
-      CLOG(LEV_DEBUG5) << "Updating preference for one of "
-                       << req->requester()->manager()->prototype()
-                       << "'s trade nodes:";
-      CLOG(LEV_DEBUG5) << "   preference: " << a.unode()->prefs[a];
-
-      graph->AddArc(a);
+      return;
+    } else if (pref == 0) {
+      std::stringstream ss;
+      ss << "0-valued preferences have been deprecated. "
+         << "Please make preference value positive."
+         << "This message will go away in before the next release (1.4).";
```

Comment:
Agreed and done

On Tue, Aug 25, 2015 at 1:06 PM, Robert Carlsen <notifications@github.com>
wrote:

> In src/exchange_translator.h
> <https://github.com/cyclus/cyclus/pull/1121#discussion_r37898197>:
>
> > -
> > -      a.unode()->prefs[a] = pref;  // request node is a.unode()
> > -      int n_prefs = a.unode()->prefs.size();
> > -
> > -      CLOG(LEV_DEBUG5) << "Updating preference for one of "
> > -                       << req->requester()->manager()->prototype()
> > -                       << "'s trade nodes:";
> > -      CLOG(LEV_DEBUG5) << "   preference: " << a.unode()->prefs[a];
> > -
> > -      graph->AddArc(a);
> > +      return;
> > +    } else if (pref == 0) {
> > +      std::stringstream ss;
> > +      ss << "0-valued preferences have been deprecated. "
> > +         << "Please make preference value positive."
> > +         << "This message will go away in before the next release (1.4).";
>
> I thought we were going to keep this loud failur for 1 release. As of now,
> it has never been in a release. I think only having the error in the tree
> for a couple weeks might not be enough for me to be confident I've fixed
> all my archetypes and input files. Thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cyclus/pull/1121/files#r37898197>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192